File: src/main/scala/com/gravity/goose/extractors/AdditionalDataExtractor.scala
package com.gravity.goose.extractors

import org.jsoup.nodes.Element
import java.util.Map

/**
* Implement this abstract class to extract anything not currently contained within the {@link com.jimplush.goose.Article} class
*/
class AdditionalDataExtractor extends Extractor[Map[String, String]] {
  def extract(rootElement: Element): Map[String, String] = {
    null
  }
}

File: src/main/java/com/jimplush/goose/extractors/AdditionalDataExtractor.java
package com.jimplush.goose.extractors; /**
 * Created by IntelliJ IDEA.
 * User: robbie
 * Date: 5/19/11
 * Time: 9:57 PM
 */

import org.jsoup.nodes.Element;

import java.util.Map;

/**
 * Implement this abstract class to extract anything not currently contained within the {@link com.jimplush.goose.Article} class
 */
public abstract class AdditionalDataExtractor implements Extractor<Map<String,String>> {

  public abstract Map<String, String> extract(Element rootElement);
  
}


